src/nanovg_pick.c

```diff
@@ -0,0 +1,1742 @@
+	
+	float const* points = &ps->points[seg->firstPoint * 2];
+	
+	switch(seg->type)
+	{
+		case NVG_LINETO:
+			{
+				d[0] = PosX(1) - PosX(0);
+				d[1] = PosY(1) - PosY(0);
+				
+				nvg__normalize(&d[0], &d[1]);
+			}
+			break;
+		case NVG_BEZIERTO:		
+			{
+				float omt = 1.0f - t;
+				float omt2 = omt * omt;
+				float t2 = t * t;
+
+				d[0] =		3.0f * omt2 * (PosX(1) - PosX(0))
+						+	6.0f * omt * t * (PosX(2) - PosX(1))
+						+	3.0f * t2 * (PosX(3) - PosX(2));	
+				d[1] =		3.0f * omt2 * (PosY(1) - PosY(0))
+						+	6.0f * omt * t * (PosY(2) - PosY(1))
+						+	3.0f * t2 * (PosY(3) - PosY(2));
+
+				nvg__normalize(&d[0], &d[1]);				
+			}
+			break;
+	}
+	
+	#undef PosX
+	#undef PosY
+}
+
+static void nvg__pickSubPathAddFillSupports(NVGpickScene* ps, NVGpickSubPath* psp)
+{
+	NVGsegment* seg = NULL;
+	NVGsegment* segments = &ps->segments[psp->firstSegment];
+	const float* points = NULL;
+	int s;
+	for (s = 0; s < psp->nsegments; ++s)
+	{
+		seg = &segments[s];
+		points = &ps->points[seg->firstPoint * 2];
+		
+		if (seg->type == NVG_LINETO)
+		{
+			nvg__initBounds(seg->bounds);
+			nvg__expandBounds(seg->bounds, points, 2);
+		}
+		else
+		{
+			nvg__bezierBounds(points, seg->bounds);
+		}
+	}
+}
+
+static void nvg__pickSubPathAddStrokeSupports(NVGpickScene* ps, NVGpickSubPath* psp, float strokeWidth, int lineCap, int lineJoin, float miterLimit)
+{
+	int closed = psp->closed;
+	const float* points = ps->points;
+
+	NVGsegment* prevseg = NULL;
+	NVGsegment* seg = NULL;
+	NVGsegment* segments = &ps->segments[psp->firstSegment];
+	int nsegments = psp->nsegments;
+	
+	prevseg = psp->closed ? &segments[psp->nsegments - 1] : NULL;
+	
+	int ns = 0; // nsupports
+	float supportingPoints[32];
+	int firstPoint, lastPoint;
+
+	if (closed == 0)
+	{
+		segments[0].flags |= NVG_PICK_CAP;
+		segments[nsegments - 1].flags |= NVG_PICK_ENDCAP;
+	}
+	
+	int s;	
+	for (s = 0; s < nsegments; ++s)
+	{
+		seg = &segments[s];
+		nvg__initBounds(seg->bounds);
+		
+		firstPoint = seg->firstPoint * 2;
+		lastPoint = firstPoint + ((seg->type == NVG_LINETO) ? 2 : 6);
+		
+		ns = 0;
+		
+		// First two supporting points are either side of the start point
+		supportingPoints[ns++] = points[firstPoint] - seg->startDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[firstPoint+1] + seg->startDir[0] * strokeWidth;
+	
+		supportingPoints[ns++] = points[firstPoint] + seg->startDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[firstPoint+1] - seg->startDir[0] * strokeWidth;
+
+		// Second two supporting points are either side of the end point
+		supportingPoints[ns++] = points[lastPoint] - seg->endDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[lastPoint+1] + seg->endDir[0] * strokeWidth;
+	
+		supportingPoints[ns++] = points[lastPoint] + seg->endDir[1] * strokeWidth;
+		supportingPoints[ns++] = points[lastPoint+1] - seg->endDir[0] * strokeWidth;
+
+//		NVG_PICK_DEBUG_VECTOR_SCALE(&points[firstPoint], seg->startDir, 10);
```

Comment:
Could put these inside the debug define, or add comment when they should be uncommented, or remove.